### PR TITLE
Add initial IamSlice suggestion

### PR DESCRIPTION
**Pull request by coroa**
This would be a possible way to add an IamSlice feature proposed in #630 .

# Please confirm that this PR has done the following:

- [ ] Tests Added
- [ ] Documentation Added
- [ ] Name of contributors Added to AUTHORS.rst
- [ ] Description in RELEASE_NOTES.md Added

## Adding to RELEASE_NOTES.md (remove section after adding to RELEASE_NOTES.md)

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

# Description of PR

Please describe the changes introduced by this PR.

### CODE REVIEW

**Comment by danielhuppmann** [comment]
I don't feel sufficiently qualified to do a proper review of the implementation, so I wrote a few tests and made some minor modifications - see #642 and #643

**Comment by coroa** [comment]
Hmm ... that's a pickle. Not sure who could do a review. For me this was basically just trying to get something out to start a discussion.

Undesirable properties (i am aware of):
- `__getitem__` looses meta. it might need a couple of extra postprocessing lines, similar to what is currently in filter (maybe that should be factored out into a utility function). But we could also decide to post-pone the getitem implementation.
- The slice does not have the same dimension categorisation information as the full dataframe. ie its info can only talk broadly about dimensions, instead of differentiating between meta, timeseries and others.

**Comment by danielhuppmann** [comment]
> * `__getitem__` looses meta. it might need a couple of extra postprocessing lines, similar to what is currently in filter (maybe that should be factored out into a utility function). But we could also decide to post-pone the getitem implementation.
> * The slice does not have the same dimension categorisation information as the full dataframe. ie its info can only talk broadly about dimensions, instead of differentiating between meta, timeseries and others.

I don't think that this is a problem. I'd even remove the `__repr__` and `info` methods in a first implementation.

File: tests/test_unfccc.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 INDEX_ARGS = dict(model="UNFCCC", scenario="Data Inventory")
 
 
-def test_unfccc_tier1():
+def _test_unfccc_tier1():
```
Comment (danielhuppmann) [inline tests/test_unfccc.py:14]: This was a quickfix to make unit-tests pass, already correctly fixed via #647 
```suggestion
def test_unfccc_tier1():
```

**Comment by danielhuppmann** [comment]
closing in favor of #657

**Pull request closed without merging**